explore/realspace.py

```diff
@@ -105,23 +110,25 @@ def volume(self):
         raise NotImplementedError()
 
     def sample(self, density):
-        # type: (float) -> np.ndarray[N], np.ndarray[N, 3]
+        # type: (float) -> Tuple[np.ndarray[N], np.ndarray[N, 3]]
```

Comment:
np.ndarray is not a parameterisable class, should be Tuple[np.ndarray, np.ndarray], as much as we might not like it

Reply:
The downside is that it breaks type inferences